bindings/node/test/autoEncrypter.test.js

```diff
@@ -110,6 +110,92 @@ describe('AutoEncrypter', function () {
     sandbox.restore();
   });
 
+  describe('#constructor', function () {
+    // This case is driver 4.9.0 and higher.
+    context('when mongodb exports BSON', function () {
+      context('when a bson option is provided', function () {
+        const bson = Object.assign({}, BSON);
+        const encrypter = new AutoEncrypter(
+          {},
+          {
+            bson: bson,
+            kmsProviders: {
+              local: { key: Buffer.alloc(96) }
+            }
+          }
+        );
+
+        it('uses the bson option', function () {
+          expect(encrypter._bson).to.equal(bson);
+        });
+      });
+
+      context('when a bson option is not provided', function () {
+        const encrypter = new AutoEncrypter(
+          {},
+          {
+            kmsProviders: {
+              local: { key: Buffer.alloc(96) }
+            }
+          }
+        );
+
+        it('uses the mongodb exported BSON', function () {
+          expect(encrypter._bson).to.equal(BSON);
+        });
+      });
```

Comment:
Should we add a test confirming that it never uses the BSON defined on the topology?  I think we explicitly test this, because if we ever did accidentally start using the client defined on the topology again, it would throw since these tests pass in an empty object for the client.  But it might be nice to explicitly document it with a test, something like

```typescript
it('should never use the BSON defined on the topology', function() {
  const client = { topology: { BSON: 'bson' } };
  const encrypter = new AutoEncrypter(....)
  expect(BSON).not.to.equal('bson')
});
```

or maybe

```typescript
it('should never use the BSON defined on the topology', function() {
  expect(() => new AutoEncrypter({}, ....)).not.to.throw(<error>);
});

```

Reply:
I chose the later for both tests.